internal/pkg/service/buffer/store/file_test.go

```diff
@@ -0,0 +1,223 @@
+package store
+
+import (
+	"context"
+	"testing"
+
+	"github.com/benbjohnson/clock"
+	"github.com/keboola/go-client/pkg/storageapi"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model/column"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
+)
+
+func TestStore_CreateFile(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	store := newStoreForTest(t)
+
+	receiverKey := key.ReceiverKey{ProjectID: 1000, ReceiverID: "github"}
+	exportKey := key.ExportKey{ExportID: "github-issues", ReceiverKey: receiverKey}
+	fileID := clock.NewMock().Now().UTC()
+	fileKey := key.FileKey{FileID: fileID, ExportKey: exportKey}
+	file := model.File{
+		FileKey: fileKey,
+		Mapping: model.Mapping{
+			MappingKey: key.MappingKey{
+				ExportKey:  exportKey,
+				RevisionID: 1,
+			},
+			TableID: model.TableID{
+				Stage:  "in",
+				Bucket: "bucket",
+				Table:  "table",
+			},
+			Incremental: false,
+			Columns: []column.Column{
+				column.Body{},
+			},
+		},
+		StorageResource: &storageapi.File{ID: 1, Name: "file1"},
+	}
+	_, err := store.createFileOp(ctx, file).Do(ctx, store.client)
+	assert.NoError(t, err)
+
+	// Check keys
+	etcdhelper.AssertKVs(t, store.client, `
+<<<<<
+file/1000/github/github-issues/1970-01-01T00:00:00.000Z
+-----
+{
+  "projectId": 1000,
+  "receiverId": "github",
+  "exportId": "github-issues",
+  "fileId": "1970-01-01T00:00:00Z",
+  "mapping": {
+    "projectId": 1000,
+    "receiverId": "github",
+    "exportId": "github-issues",
+    "revisionId": 1,
+    "tableId": {
+      "stage": "in",
+      "bucketName": "bucket",
+      "tableName": "table"
+    },
+    "incremental": false,
+    "columns": [
+      {
+        "type": "body"
+      }
+    ]
+  },
+  "storageResource": {
+    "id": 1,
+    "created": "0001-01-01T00:00:00Z",
+    "name": "file1",
+    "url": "",
+    "provider": "",
+    "region": "",
+    "maxAgeDays": 0,
+    "absUploadParams": {
+      "blobName": "",
+      "accountName": "",
+      "container": "",
+      "absCredentials": {
+        "SASConnectionString": "",
+        "expiration": "0001-01-01T00:00:00Z"
+      }
+    },
+    "gcsUploadParams": {
+      "projectId": "",
```

Comment:
Mozno by sa nam tam hodil pointer aby sa dala odlisit prazdna hodnota (a aby sa to potom zbytocne neukladalo, ale funkcne je to aj takto. Mala by to byt iba drobna zmena v `go-client`.